srcpkgs/avahi-discover/template

```diff
@@ -1,42 +1,53 @@
-# ADDITIONAL DEPENDENCIES LIKE GTK3 AND PyGI.
+# WARNING: THIS TEMPLATE MUST BE SYNCHRONIZED WITH AVAHI.
+# IT IS SPLIT TO AVOID DEPENDENCY CYCLES WITH GTK3.
 #
 pkgname=avahi-discover
-version=0.7
-revision=8
+version=0.8
+revision=2
 wrksrc="avahi-${version}"
 build_style=gnu-configure
 build_helper="gir"
 configure_args="--disable-qt3 --disable-qt4 --disable-mono --disable-monodoc
  --disable-doxygen-doc --enable-compat-libdns_sd --enable-compat-howl
  --with-xml=expat --with-avahi-user=avahi --with-avahi-group=avahi
  --with-avahi-priv-access-group=network --with-autoipd-user=avahi
- --with-autoipd-group=avahi --with-distro=none --disable-xmltoman
+ --with-autoipd-group=avahi --with-distro=none --enable-xmltoman
  --disable-dbm --without-systemdsystemunitdir
+ --with-dbus-system-address=unix:path=/run/dbus/system_bus_socket
  --disable-static ssp_cv_lib=no --enable-python
  --enable-glib --enable-gobject --enable-gtk --enable-gtk3
  $(vopt_enable gir introspection) $(vopt_enable gir pygobject)"
-short_desc="Service discover user interface for avahi"
-maintainer="Orphaned <orphan@voidlinux.org>"
+short_desc="Service discovery user interface for avahi"
+maintainer="Andrew J. Hesford <ajh@sideband.org>"
 license="LGPL-2.1-or-later"
 homepage="https://github.com/lathiat/avahi"
-distfiles="${homepage}/releases/download/v${version}/avahi-${version}.tar.gz"
-checksum=57a99b5dfe7fdae794e3d1ee7a62973a368e91e414bd0dfa5d84434de5b14804
+### NOTE: Avahi 0.8 release tarball is broken; see avahi template for info
+distfiles="${homepage}/releases/download/v${version}/avahi-${version}.tar.gz
+ https://raw.githubusercontent.com/lathiat/avahi/v${version}/service-type-database/build-db>build-db.txt"
+checksum="060309d7a333d38d951bc27598c677af1796934dbd98e1024e7ad8de798fedda
+ 83edd888ccc464ab032134a45b7ded3bc43cb1dda2cd921ff47b01949d98ed7c"
+skip_extraction="build-db.txt"
 
-hostmakedepends="automake libtool pkg-config intltool
- python-dbus gobject-introspection glib-devel"
-makedepends="dbus-devel libcap-devel libdaemon-devel gdbm-devel
- python-dbus gtk+-devel gtk+3-devel $(vopt_if gir python-gobject-devel)"
-depends="avahi>=${version} avahi-python>=${version}
- $(vopt_if gir 'gtk+3 python-gobject') desktop-file-utils"
-pycompile_module="avahi_discover"
+python_version=3
+
+hostmakedepends="automake libtool pkg-config intltool gettext-devel
+ python3-dbus gobject-introspection glib-devel qt5-host-tools xmltoman"
+makedepends="dbus-devel libcap-devel libdaemon-devel gdbm-devel libevent-devel
+ python3-dbus gtk+-devel gtk+3-devel qt5-devel $(vopt_if gir python3-gobject-devel)"
+depends="avahi>=${version} avahi-python3>=${version}_${revision}
+ $(vopt_if gir 'gtk+3 python3-gobject') desktop-file-utils"
 
 # Package build options
 build_options="gir"
 build_options_default="gir"
 
+post_extract() {
+	# Avahi 0.8 release accidentally left this out of the tarball
+	cp ${XBPS_SRCDISTDIR}/${pkgname}-${version}/build-db.txt service-type-database/build-db
```

Comment:
I think you are leaving `base-db.txt` around in `masterdir/builddir` here and in the other template which will not be cleaned up.
I think it should be moved into the right directory, instead of copying from `XBPS_SRCDISTDIR`.

Reply:
The `skip_extraction` means that `build-db.txt` is never copied from `XBPS_SRCDISTDIR` during fetch, so there are no artifacts in `masterdir/builddir`. The copy puts these files within the `avahi` build directory, which will be completely removed after packaging.